src/backend/distributed/transaction/relation_access_tracking.c

```diff
@@ -227,10 +227,12 @@ RecordRelationAccessBase(Oid relationId, ShardPlacementAccessType accessType)
 
 	/*
 	 * If a relation is partitioned, record accesses to all of its partitions as well.
-	 * We prefer to use PartitionedTableNoLock() because at this point the necessary
-	 * locks on the relation has already been acquired.
+	 *
+	 * TODO: We should use PartitionedTableNoLock() and PartitionTableNoLock() here
+	 * after making sure all of the calls that end up here acquire proper locks.
+	 * See https://github.com/citusdata/citus/issues/2888.
 	 */
-	if (PartitionedTableNoLock(relationId))
+	if (PartitionedTable(relationId))
```

Comment:
@onderkalaci @marcocitus what do you think postponing a proper solution for this to 9.0? I'm suggesting this because we are tight on time for this release and these issues are issues we had in previous releases too, but were just unveiled by postgres 12's assertions, so no new issues.

If we get free time before the release we can in look into this more properly though.

Reply:
Actually, as of #2866 we're only calling `RecordRelationAccessBase ()` for reference tables. So, they cannot be partitioned tables anyway -- for now. 

And, for reference tables the performance bottleneck is not usually related anything happening on the coordinator, plus we're only acquiring `AccessShareLock`.

I'm OK to keep as-is, or even remove the logic for partition/partitioned tables here.

Reply:
Removed this checks and added an assert.